src/compiler/sys.ts

```diff
@@ -494,9 +501,18 @@ namespace ts {
                     }
                 },
                 createHash(data) {
-                    const hash = _crypto.createHash("md5");
-                    hash.update(data);
-                    return hash.digest("hex");
+                    let hash: any;
+                    if (_crypto) {
+                      hash = _crypto.createHash("md5");
+                      hash.update(data);
+                      hash.digest("hex");
+                    }
+                    else {
+                      // djb2 hashing algorithm: http://www.cse.yorku.ca/~oz/hash.html
+                      const chars = data.split("").map(str => str.charCodeAt(0));

```

Comment:
Can you make this a exported function with ```/*@internal*/``` and also use it at https://github.com/Microsoft/TypeScript/blob/master/src/compiler/watch.ts#L645 in the ```system.createHash``` ```undefined``` case.